File: 767-prime-number-of-set-bits-in-binary-representation/prime-number-of-set-bits-in-binary-representation.py
class Solution:
    def is_prime(self, n: int) -> bool:
        if n <= 1:
            return False
        if n == 2:
            return True
        if n % 2 == 0:
            return False
        for i in range(3, int(sqrt(n)) + 1, 2):
            if n % i == 0:
                return False
        return True

    def count_set_bits(self, n: int) -> int:
        count = 0
        while n:
            count += n & 1
            n >>= 1
        return count

    def countPrimeSetBits(self, left: int, right: int) -> int:
        count = 0
        for num in range(left, right + 1):
            set_bits_count = self.count_set_bits(num)
            if self.is_prime(set_bits_count):
                count += 1
        return count
        

File: 767-prime-number-of-set-bits-in-binary-representation/prime-number-of-set-bits-in-binary-representation.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
var countPrimeSetBits = function(left, right) {
    /* let result = 0;

    for (let num = left; num <= right; num++) {
        let binary = num.toString(2);

        let count = 0;
        for (let char of binary) {
            if (char === '1') {
                count ++;
            }
        }

        if (isPrime(count)) {
            result ++
        }
    }

    return result;

    function isPrime(n) {
        if (n <= 1) return false;
        for (let i = 2; i * i <= n; i++) {
            if (n % i === 0) {
                return false;
            }
        }
        return true;
    } */

    let result = 0;

    for (let num = left; num <= right; num++) {
        let count = countBits(num);

        if (isPrime(count)) {
            result++;
        }
    }
    return result;

    /* function countBits(num) {
        let count = 0;
        while (num > 0) {
            count += num & 1;
            num = num >> 1;
        }
        return count;
    } */

    function countBits(num) {
        let count = 0;

        while (num > 0) {
            num = num & (num - 1);
            count++;
        }
        return count;
    }

    function isPrime(n) {
        if (n <= 1) return false;

        for (let i = 2; i * i <= n; i++) {
            if (n % i === 0) return false;
        }
        return true;
    }
};